nixos/modules/programs/shadow.nix

```diff
@@ -58,6 +58,17 @@ in
 
   options = {
 
+    security.loginDefs.extraConfig = mkOption {
+      default = "";
+      type = types.lines;
```

Comment:
Why not use a rfc42 style type here?

Reply:
And therefore also melt the hard coded [login.defs](https://github.com/NixOS/nixpkgs/blob/9f1c3c7dad56a197a55def1963d0737ae74f0b7b/nixos/modules/programs/shadow.nix#L19-L44) into an ini settings format with `pkgs.formats.ini`? This would also tackle your other suggestion with the `CHFN_RESTRICT` option.

Reply:
ok, not ini I wrote a custom toKeyValue generator to reflect the login.defs config spec.